### DISCUSSION TITLE
port block test ....does not resolve

### DISCUSSION THREAD

**Discussion by mel-svg** [bol-van/zapret#309] [Q&A]
Доброго времени суток. Озадачился оживлением Youtube и накатил OpenWrt на роутер, перед установкой прогоняю blockcheck.sh и получаю следующее: 

```
* port block tests ipv4 rutracker.org:80
ipv4 rutracker.org does not resolve

* curl_test_http ipv4 rutracker.org
- checking without DPI bypass
UNAVAILABLE code=6

* port block tests ipv4 rutracker.org:443
ipv4 rutracker.org does not resolve

* curl_test_https_tls12 ipv4 rutracker.org
- checking without DPI bypass
UNAVAILABLE code=6

* port block tests ipv6 rutracker.org:80
ipv6 rutracker.org does not resolve

* curl_test_http ipv6 rutracker.org
- checking without DPI bypass
UNAVAILABLE code=6

* port block tests ipv6 rutracker.org:443
ipv6 rutracker.org does not resolve

* curl_test_https_tls12 ipv6 rutracker.org
- checking without DPI bypass
UNAVAILABLE code=6
```

Это приговор?

и ещё такой: вопрос перед началом проверки пишет:

```
installed curl version does not support TLS 1.3 . tests disabled.

installed curl version does not support http3 QUIC. tests disabled.
```

curl вроде последний доступный. Насколько это влияет на результаты теста?

И самое главное имеет ли смысл устанавливать для обхода замедления Youtube, если на тестах на заблокированных сайтах пишет про порт? Мне главное именно Youtube, с остальными буду разбираться по другому

**Comment by bol-van** [comment]
Это значит у вас православный DNS, который исключает заблокированные домены
Менять DNS. Если сторонние DNS перехватываются (провайдер домру), ставить doh

**Reply by mel-svg** [reply]
я вроде подкинул quad9
dnsleaktest показывает

```
IP 	Hostname 	ISP 	Country
109.200.195.200 	res300.qarn2.rrdns.quad9.net. 	i3D.net 	Spånga, Sweden
109.200.195.202 	res700.qarn2.rrdns.quad9.net. 	i3D.net 	Spånga, Sweden
109.200.195.203 	res701.qarn2.rrdns.quad9.net. 	i3D.net 	Spånga, Sweden 
```

**Reply by bol-van** [reply]
на роутере
ping rutacker.org
или
nslookup rutracker.org
что говорят ?
менять днс надо там, где выполняется zapret (на роутере, например)

**Reply by mel-svg** [reply]
Спасибо за подсказку. Естественно менял на роутере. И через dnscrypt-proxy2 и через https-dns-proxy. В обоих случаях тот же rutracker с роутера не пинговался (сразу писало bad request или что-то там со словом bad), а при этом dnsleaktest показывал выбранного провайдера днс (quad9, google, cloudflare, adguard, etc. всегда показывал верно). Пинга не было. Сбрасывал настройки, настраивал заново. Результат тот же. Плюнул, перепрошил роутер, поставил https-dns-proxy, выбрал провайдера dns и о чудо rutracker пингуется. Прошел blockcheck.sh на тот же rutacker.org, ввел предложенную стратегию nfqws. Как результат Youtube работает, кроме официального приложения на Android TV, но меня SmartTube, устраивает более чем.
Но с dns все равно что-то не так. Часть заблокированных сайтов пингуется, часть нет. Думал еще раз пройти blockcheck.sh на другие адреса, ввел instagram и на каждый запрос получается Connection timed out after 2002 milliseconds, попробовал его пингануть, пинга нет (просто тишина, никакого ответа пока не прервешь), а через час попробуешь и пингуется, а следующая попытка неудачная и так с многими заблоченными сайтами. На rutracker сейчас кстати даже с остановленным zapretом ходит.
Сегодня проверил dnsleaktest, а он показывает dns моего домашнего провайдера, rutracker кстати пинговался в этот момент. Перезапустил службу https-dns-proxy то же самое, перезагрузил роутер тоже самое. Добавил еще дного провайдера, dnsleaktest показывает трех мною выбранных. И как к этому относиться не понятно
Пока все это писал, периодически пробовал пинговать instagram. Из 8 попыток 2 удачные...

**Reply by bol-van** [reply]
Учитесь сканировать порты
Проверить порт можно netcat , ncat от nmap, самим nmap, телнетом и другими средствами
Если хост не конектит на порт, все остальные проверки не имеют смысла и только теряют время

**Comment by bol-van** [comment]
curl от openwrt ограничен в возможностях
если нужна нормальная версия, ищите бинарики для вашей архитектуры тут https://curl.se/download.html
но они весят по 8 мб, так что надо достаточно места иметь или extroot
или проверять с компа, а потом в стратегиях уменьшать ттл на 1, если он там есть
но проверка с компа может не выявить некоторые стратегии, доступные только с самого роутера, а не за ним
